app/containers/message/Components/Audio/Slider.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { LayoutChangeEvent, View, TextInput } from 'react-native';
+import { PanGestureHandler } from 'react-native-gesture-handler';
+import Animated, {
+	SharedValue,
+	runOnJS,
+	useAnimatedGestureHandler,
+	useAnimatedProps,
+	useAnimatedStyle,
+	useDerivedValue,
+	useSharedValue
+} from 'react-native-reanimated';
+
+import styles from './styles';
+import { useTheme } from '../../../../theme';
+
+const AnimatedTextInput = Animated.createAnimatedComponent(TextInput);
+
+interface ISlider {
+	duration: SharedValue<number>;
+	currentTime: SharedValue<number>;
+	loaded: boolean;
+	onChangeTime: (time: number) => Promise<void>;
+}
+
+const BUTTON_HIT_SLOP = {
+	top: 8,
+	right: 8,
+	bottom: 8,
+	left: 8
+};
+
+const Slider = ({ currentTime, duration, loaded = false, onChangeTime }: ISlider) => {
+	const { colors } = useTheme();
+
+	const maxWidth = useSharedValue(1);
+	const x = useSharedValue(0);
+	const current = useSharedValue('00:00');
+	const scale = useSharedValue(1);
+	const isHandlePan = useSharedValue(false);
+	const onEndGestureHandler = useSharedValue(false);
+	const isTimeChanged = useSharedValue(false);
+
+	const styleLine = useAnimatedStyle(() => ({
+		width: x.value,
+		zIndex: 2
+	}));
+
+	const styleThumb = useAnimatedStyle(() => ({
+		transform: [{ translateX: x.value }, { scale: scale.value }]
+	}));
+
+	const onLayout = (event: LayoutChangeEvent) => {
+		const { width } = event.nativeEvent.layout;
+		maxWidth.value = width - 12;
+	};
+
+	const gestureHandler = useAnimatedGestureHandler({
+		onStart: (_, ctx: any) => {
+			ctx.startX = x.value;
+			isHandlePan.value = true;
+		},
+		onActive: (event, ctx: any) => {
+			const moveInX: number = ctx.startX + event.translationX;
+			if (moveInX < 0) {
+				x.value = 0;
+			} else if (moveInX > maxWidth.value) {
+				x.value = maxWidth.value;
+			} else {
+				x.value = moveInX;
+			}
+			isTimeChanged.value = true;
+			scale.value = 1.3;
+		},
+		onEnd: () => {
+			scale.value = 1;
+			isHandlePan.value = false;
+			onEndGestureHandler.value = true;
+		}
+	});
+
+	const wrapper = async (time: number) => {
+		await onChangeTime(Math.round(time * 1000));
+		onEndGestureHandler.value = false;
+	};
+
+	useDerivedValue(() => {
+		if (onEndGestureHandler.value) {
+			runOnJS(wrapper)(currentTime.value);
+		}
+	});
+
+	const formatTime = (ms: number) => {
+		'worklet';
```

Comment:
Add the link to why you are using this property.

Reply:
https://docs.swmansion.com/react-native-reanimated/docs/2.x/fundamentals/worklets/

We need the "worklet" because this function is being called inside the worklet useDerivedValue. The function is to don't repeat the same code twice at lines 107 and 111. 

If there isn't an explicit "worklet" at formatTime function, this function will only run on the JS side, not on the UI Thread created, and won't be capable of running synchronously with the useDerivedValue

Reply:
You didn't understand my point, you didn't leave any information in the CODE about why you used this property, a person who doesn't understand the use of this prop when reading the code won't understand why it was used, when adding very specific things try always add comments to the code.

Reply:
Before merging this PR, add this information to the code.